sklearn/callback/_callback_context.py

```diff
@@ -83,21 +148,87 @@ def _from_parent(cls, parent_context, *, task_name, task_id, max_tasks=1):
         new_ctx = cls.__new__(cls)
 
         new_ctx._callbacks = parent_context._callbacks
-        new_ctx._estimator_name = parent_context._estimator_name
+        new_ctx.estimator_name = parent_context.estimator_name
         new_ctx._estimator_depth = parent_context._estimator_depth
-
-        new_ctx._task_node = TaskNode(
-            task_name=task_name,
-            task_id=task_id,
-            max_tasks=max_tasks,
-            estimator_name=new_ctx._estimator_name,
-        )
+        new_ctx.task_name = task_name
+        new_ctx.task_id = task_id
+        new_ctx.max_tasks = max_tasks
+        new_ctx.parent = None
+        new_ctx.children_map = {}
+        new_ctx.max_subtasks = 0
+        new_ctx.prev_estimator_name = None
+        new_ctx.prev_task_name = None
 
         # This task is a subtask of another task of a same estimator
-        parent_context._task_node._add_child(new_ctx._task_node)
+        parent_context._add_child(new_ctx)
 
         return new_ctx
 
+    def _to_dict(self):
+        # function that returns attributes and parent as parent's _to_dict
+        context_dict = {
+            "estimator_name": self.estimator_name,
+            "depth": self.depth,
+            "task_name": self.task_name,
+            "task_id": self.task_id,
+            "max_tasks": self.max_tasks,
+            "max_subtasks": self.max_subtasks,
+            "prev_estimator_name": self.prev_estimator_name,
+            "prev_task_name": self.prev_task_name,
+            "parent": None if self.parent is None else self.parent._to_dict(),
+        }
+        return context_dict
+
+    @property
+    def depth(self):
+        """The depth of this node in the task tree."""
+        return 0 if self.parent is None else self.parent.depth + 1
+
+    @property
+    def path(self):
+        """List of all the nodes in the path from the root to this node."""
+        return (
+            [self._to_dict()]
+            if self.parent is None
+            else self.parent.path + [self._to_dict()]
+        )
```

Comment:
I think this is not used anymore since you implemented a dedicated helper for the progressbar. Let's remove it